some tests which require internet access fail, is it enough to disable them, or should I try to work around that?
I guess fetching the required things beforehand, somehow making the tests pick up on, that the files are already there.

```
Running phase: checkPhase
@nix { "action": "setPhase", "phase": "checkPhase" }
ok  	go.minekube.com/gate/cmd/gate	0.006s
ok  	go.minekube.com/gate/pkg/command	0.003s
ok  	go.minekube.com/gate/pkg/edition/bedrock/config	0.003s
ok  	go.minekube.com/gate/pkg/edition/bedrock/geyser/floodgate	0.003s
--- FAIL: TestGeyserDownloadAPI (0.00s)
    --- FAIL: TestGeyserDownloadAPI/HEAD_request_for_metadata (0.00s)
        download_test.go:30: HEAD request failed: Head "https://download.geysermc.org/v2/projects/geyser/versions/latest/builds/latest/downloads/standalone": dial tcp: lookup download.geysermc.org on [::1]:53: read udp [::1]:42334->[::1]:53: read: connection refused
    --- FAIL: TestGeyserDownloadAPI/conditional_GET_with_If-Modified-Since (0.00s)
        download_test.go:72: HEAD request failed: Head "https://download.geysermc.org/v2/projects/geyser/versions/latest/builds/latest/downloads/standalone": dial tcp: lookup download.geysermc.org on [::1]:53: read udp [::1]:53409->[::1]:53: read: connection refused
    --- FAIL: TestGeyserDownloadAPI/conditional_GET_with_If-None-Match (0.00s)
        download_test.go:116: HEAD request failed: Head "https://download.geysermc.org/v2/projects/geyser/versions/latest/builds/latest/downloads/standalone": dial tcp: lookup download.geysermc.org on [::1]:53: read udp [::1]:44097->[::1]:53: read: connection refused
--- FAIL: TestDownloadIfNewer (0.00s)
    --- FAIL: TestDownloadIfNewer/download_new_file (0.00s)
        download_test.go:204: downloadIfNewer failed: Get "https://download.geysermc.org/v2/projects/geyser/versions/latest/builds/latest/downloads/standalone": dial tcp: lookup download.geysermc.org on [::1]:53: read udp [::1]:60674->[::1]:53: read: connection refused
    --- FAIL: TestDownloadIfNewer/check_for_updates_when_file_exists (0.00s)
        download_test.go:238: Initial download failed: Get "https://download.geysermc.org/v2/projects/geyser/versions/latest/builds/latest/downloads/standalone": dial tcp: lookup download.geysermc.org on [::1]:53: read udp [::1]:35573->[::1]:53: read: connection refused
    --- FAIL: TestDownloadIfNewer/immediate_second_check_should_not_update (0.00s)
        download_test.go:283: Initial download failed: Get "https://download.geysermc.org/v2/projects/geyser/versions/latest/builds/latest/downloads/standalone": dial tcp: lookup download.geysermc.org on [::1]:53: read udp [::1]:33314->[::1]:53: read: connection refused
FAIL
FAIL	go.minekube.com/gate/pkg/edition/bedrock/geyser/managed	2.090s
FAIL
```

--- a/pkgs/by-name/ga/gate/package.nix
+++ b/pkgs/by-name/ga/gate/package.nix
@@ -6,7 +6,7 @@
 
 let
   pname = "gate";
-  version = "0.53.0";
+  version = "0.57.1";
 in
 buildGoModule {
   inherit pname version;
@@ -15,15 +15,20 @@ buildGoModule {
     owner = "minekube";
     repo = "gate";
     tag = "v${version}";
-    hash = "sha256-wrvq2opwT4bbplUljasWmT+JF3/lS8AyzBSfyUB3nUw=";
+    hash = "sha256-G4kmXGiogl/W6SYVWZyQsQE+6YO5Yggk8K4rH+t9znE=";
   };
 
-  vendorHash = "sha256-0NcfuCZHR4QHbMNqc+ilPouie+9k7FqOG/JdNX8uO8c=";
+  vendorHash = "sha256-2ZRfvjIGUznHjn7KA20uzEpVbI7EByNUYu6xALJEUfo=";
 
   ldflags = [
     "-s"
     "-w"
   ];
+
+  # this test requires network access, therefore it should not be run
+  preCheck = ''
+    rm ./pkg/edition/bedrock/geyser/managed/download_test.go
+  '';
 
   excludedPackages = [ ".web" ];
 

